src/core/utils/EventNormalizer.ts

```diff
@@ -294,30 +294,83 @@ export class EventNormalizer {
                 // actual keypress event but trigger an input event with the
                 // LINE FEED (LF) (u000A) unicode character instead.
                 // TODO: replace this <br/> by a contextualized 'enter' event
-                this._triggerEvent('insert', '<br/>', elements);
+                this._triggerEvent(
+                    this._formatCustomEvent(
+                        'insert',
+                        {
+                            value: '<br/>',
+                        },
+                        elements,
+                    ),
+                );
             } else {
-                this._triggerEvent('insert', data, elements);
+                this._triggerEvent(
+                    this._formatCustomEvent(
+                        'insert',
+                        {
+                            value: data,
+                        },
+                        elements,
+                    ),
+                );
             }
         } else if (compiledEvent.type === 'keydown') {
             // TODO: Maybe the normalizer should not trigger keydown events:
             // - they are consistent accross browsers so no normalization needed
             // - they won't be able to be defaultPrevented after being triggered
-            this._triggerEvent('keydown', compiledEvent);
+            this._formatAndTriggerEvent('keydown', compiledEvent);
         } else {
             // Something definitely happened since some events were compiled,
             // but it appears to be currently unsupported since it did not fall
             // in any of the previous conditional branches. The listener has no
             // way to understand the change in the DOM since the normalizer does
             // not understand it either. Let it know about the inconsistence.
-            this._triggerEvent('inconsistentState', compiledEvent);
+            this._formatAndTriggerEvent('inconsistentState', compiledEvent);
         }
     }
+    /**
+     * Return a properly formatted CustomEvent, based on the given type
+     * and optional params.
+     *
+     * @param {string} type
+     * @param {object} [params]
+     * @param {Set<HTMLElement>} [elements]
+     * @returns {CustomEvent}
+     */
+    _formatCustomEvent(type: string, params = {}, elements?: Set<HTMLElement>): CustomEvent {
+        const initDict = {
+            detail: params,
+        };
+        initDict.detail['origin'] = 'EventNormalizer';
+        if (elements) {
+            initDict.detail['elements'] = elements;
+        }
+        return new CustomEvent(type, initDict);
+    }
+    /**
+     * Format a custom event and trigger it.
+     *
+     * @param {string} type
+     * @param {object} [params]
+     * @param {Set<HTMLElement>} [elements]
+     */
+    _formatAndTriggerEvent(type: string, params = {}, elements?: Set<HTMLElement>): void {
+        this._triggerEvent(this._formatCustomEvent(type, params, elements));
```

Comment:
At this stage, I think I would just rename `_triggerEvent` into `_eventCallback`, then rename `_formatAndTriggerEvent` into `_triggerEvent`, and finally move the code of `_formatCustomEvent` inside `_triggerEvent`. This would be a private function of the normalizer where it triggers a Custom Event on the callback, the fact that it has to format it first is an implementation detail IMO.